### PR TITLE
xds: add new fields to XDSConfig

### PR DESCRIPTION
**Pull request by menghanl**
Fields are added in: https://github.com/grpc/grpc-proto/pull/64

Other changes:
 - Move XDSConfig from internal to balancer
   - Later we will add a separate config for CDS balancer
 - generate service_config.pb.go and test with json generated from proto message

### CODE REVIEW

File: internal/proto/grpc_service_config/example_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/jsonpb"
+	wrapperspb "github.com/golang/protobuf/ptypes/wrappers"
+	scpb "google.golang.org/grpc/internal/proto/grpc_service_config"
+)
+
+// TestExampleMarshalToJSON is an example to print json format of xds_config.
```
Comment (dfawley) [inline internal/proto/grpc_service_config/example_test.go]: Name doesn't match.  I thought this was going to be an example without verification?

File: xds/internal/balancer/config.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	// will be disabled.  If set to the empty string, load reporting will
+	// be sent to the same server that we obtained CDS data from.
+	LrsLoadReportingServerName string
+}
+
+// UnmarshalJSON parses the JSON-encoded byte slice in data and stores it in l.
+// When unmarshalling, we iterate through the childPolicy/fallbackPolicy lists
+// and select the first LB policy which has been registered.
+func (l *XDSConfig) UnmarshalJSON(data []byte) error {
+	var val map[string]json.RawMessage
```
Comment (dfawley) [inline xds/internal/balancer/config.go]: Optional, may not be worth it: would it be easier to define the literal struct, Unmarshal into it, and then post-process:

```
type xdsConfigJSON struct {
  BalancerName string
  ChildPolicy []*loadBalancingConfig
  FallbackPolicy []*loadBalancingConfig
  EDSServiceName string
  LRSLoadReportingServerName string
}
```

File: xds/internal/balancer/config.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// MarshalJSON returns a JSON encoding of l.
+func (l *XDSConfig) MarshalJSON() ([]byte, error) {
+	return nil, nil
```
Comment (dfawley) [inline xds/internal/balancer/config.go]: Should this return an unimplemented error or grpclog.Fatal() instead?

File: internal/proto/grpc_service_config/example_test.go (hunk 1 above)
Comment (menghanl) [reply internal/proto/grpc_service_config/example_test.go]: Name fixed.

Examples don't output anything. Tests can print the json (to be copied or modified).

File: xds/internal/balancer/config.go (hunk 3 above)
Comment (menghanl) [reply xds/internal/balancer/config.go]: Return an error now.

File: xds/internal/balancer/config.go (hunk 2 above)
Comment (menghanl) [reply xds/internal/balancer/config.go]: This is a bit cleaner. Done

**Pull request merged**